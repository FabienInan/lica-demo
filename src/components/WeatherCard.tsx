import React from 'react';
import styled from 'styled-components';

const CardContainer = styled.div`
    margin: 8px;
    height: 216px;
    width: 148px;
    border-radius: 7px;
    box-shadow: 0.3px 0.3px 0.3px 1px rgba(0, 0, 0, 0.2);
    text-align: center;
`;

const Date = styled.div`
    color: gray;
    margin: 0 8px;
    line-height: 40px;
    font-size: 16px;
`;

const Temperature = styled.div`
    font-size: 24px;
`;

const WeatherIcon = styled.img`
    width: 80px;
    height: 80px;
    margin: auto;
    display: block;
`;

type WeatherCardProps = {
    date: number;
    weatherType: string;
    temperatureInCelsius: number;
}

const WeatherCard = (props: WeatherCardProps) => {
    return (
        <CardContainer>
            <Date>{props.date}</Date>
            <WeatherIcon src={require(`../assets/${props.weatherType}.png`).default} alt={props.weatherType}/>
            <Temperature>{props.temperatureInCelsius}Â°C</Temperature>
        </CardContainer>
    )
};

export default WeatherCard;