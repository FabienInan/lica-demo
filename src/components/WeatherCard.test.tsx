import React from 'react';
import WeatherCard from './WeatherCard';
import renderer from 'react-test-renderer';

it('should render properly the Weather Card', () => {
    const props = {    
        date: 1234567,
        weatherType: 'clear',
        temperatureInCelsius: 23,
    } 
    const tree = renderer.create(<WeatherCard date={props.date} weatherType={props.weatherType} temperatureInCelsius={props.temperatureInCelsius}/>).toJSON();
    expect(tree).toMatchSnapshot();
});