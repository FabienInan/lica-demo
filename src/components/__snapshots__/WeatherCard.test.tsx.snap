// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly the Weather Card 1`] = `
<div
  className="sc-bdvvtL goGkoR"
>
  <div
    className="sc-gsDKAQ gFrkXI"
  >
    1234567
  </div>
  <img
    alt="clear"
    className="sc-hKwDye jOUQVu"
  />
  <div
    className="sc-dkPtRN laGaeU"
  >
    23
    Â°C
  </div>
</div>
`;
